src/gui/editors/AutomationEditor.cpp

```diff
@@ -1314,8 +1314,9 @@ void AutomationEditor::paintEvent(QPaintEvent * pe )
 		m_leftRightScroll->setPageStep( l );
 	}
 
-	if( validPattern() )
+	if( validPattern() && GuiApplication::instance()->automationEditor()->hasFocus() )
```

Comment:
Aren't you already inside the `AutomationEditor` instance?

In this case wouldn`t simply `hasFocus()` work?